### DISCUSSION TITLE
Is there a way in ibis to see generated SQL without opening a cursor to the backend

### DISCUSSION THREAD

**Discussion by sarath-mec** [ibis-project/ibis#8401] [Q&A] [answered]
I am generating some scripts by binding data to SQL generated using IBIS. In this process I understood that the following two options connect to DB by opening a Cursor.
I am trying a SQL Server Backend
```
expr = source.filter([source['CurrencyKey']==3])

expr.compile()

ibis.show_sql(expr)
```

Both connect to backend and not scalable. Is there a easy way to see the transpiled SQL without connecting to DB

**Comment by cpcloud** [comment, marked as answer]
As of 9.2 you should be able to do this. The compilers have been moved out of the dependency import path. If you're still unable to compile Ibis expressions without installing backend dependences and opening a cursor to the backend, please open a bug report (not a discussion).
